### PR TITLE
Implicit variables to @expression s

### PR DESCRIPTION
**Pull request by EliLazarus**
Refactoring to take out implicit variables, and replace with @expressions.

Articulately generating implicit variables and passing them to JuMP->PATH as variables and constraints is making the number of equations exponentially bigger, and we assume significantly slower to (build and?) solve.
Instead of generating independent variables/constraints, we will use JuMP's @expressions to generate equivalent sub-expressions that will be used directly in the main equations, vastly reducing the number of variables and equations being solved.
This branch builds from the unmerged Add/fix-WiNnat-test branch and #144 PR.

### CODE REVIEW

**Comment by EliLazarus** [comment]
Something is off with the AlgebraicWrapper and algebraic print of the model, so for now I'm commented those tests out (I like to see some green checks). 
```
avm2 = algebraic_version(m)
# TODO repr(MIME("text/plain"), avm2)
# TODO repr(MIME("text/latex"), avm2).
```
One value in one counterfactual for one model test changed enough to fail at 1e-8 decimals so I added an atol.
[output tax to 0.1 in the "TWOBYTWO (functional version , with non-1 prices in Ouput)" model test in test_twobytwo_Price]

The counterfactual setting both output taxes to 0.9 now needs a consumer value set and fixed.
[otaxa and otaxb to 0.9 in the "TWOBYTWOwOTax_IndCons" model in test_twobytwo_wTaxes]. That was already needed for the counterfactual with just otaxa set to 0.9 for the same model.

**Comment by EliLazarus** [comment]
I pushed through the code coverage test fails.
I think this is it, at least it's a working version with the @expression s. I still have questions about whether we're still building more than we need to, but that can be subsequent work.

**Review by mitchphillipson — COMMENTED** [review]
I will get to this one, it may take until Monday morning. There are more changes here that I want to dive into.

**Comment by mitchphillipson** [comment]
This includes the changes from the previous PR. You should PR in from Main to JuMPexpressions to isolate the new changes

File: src/build/build_implicitconstraints.jl
Comment (mitchphillipson) [inline src/build/build_implicitconstraints.jl:1]: File name should probably change to `build_implicitexpressions.jl` or something. Very minor

**Review by mitchphillipson — APPROVED** [review]

**Pull request merged**